strawberryfields/ops.py

```diff
@@ -542,7 +542,12 @@ def __init__(self, a=0.0, p=0.0):
 
     def _apply(self, reg, backend, **kwargs):
         p = self.p[0] * pf.exp(1j * self.p[1])
-        z = par_evaluate(p)
+
+        if np.any(self.p[1] != 0):
```

Comment:
Since this pattern is repeated several times, and the only clear explanation for what it's doing is in a resolved github comment, I think it would be worthwhile to pull out into a separate helper function (with docstring which explains behaviour under non-batched and batched operation)

Reply:
Have done!